clusterloader2/framework/util/util.go

```diff
@@ -0,0 +1,152 @@
+		_, err := c.CoreV1().Namespaces().Create(&apiv1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
+		if err == nil || apierrs.IsAlreadyExists(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to create namespace with non-retriable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(createFunc)
+
+}
+
+func DeleteNamespace(c clientset.Interface, namespace string) error {
+	deleteFunc := func() (bool, error) {
+		err := c.CoreV1().Namespaces().Delete(namespace, nil)
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to delete namespace with non-retriable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(deleteFunc)
+}
+
+func CreateObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind, obj *unstructured.Unstructured) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	obj.SetName(name)
+	createFunc := func() (bool, error) {
+		_, err := dynamicClient.Resource(gvr).Namespace(namespace).Create(obj)
+		if err == nil || apierrs.IsAlreadyExists(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to create object with non-retriable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(createFunc)
+}
+
+func DeleteObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	deleteFunc := func() (bool, error) {
+		err := dynamicClient.Resource(gvr).Namespace(namespace).Delete(name, nil)
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to delete object with non-retriable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(deleteFunc)
+}
+
+func GetObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind) (*unstructured.Unstructured, error) {
+	var obj *unstructured.Unstructured
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	getFunc := func() (bool, error) {
+		var err error
+		obj, err = dynamicClient.Resource(gvr).Namespace(namespace).Get(name, metav1.GetOptions{
+			IncludeUninitialized: true,
+		})
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to retrieve object with non-retriable error: %v", err)
+	}
+	if err := RetryWithExponentialBackOff(getFunc); err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
+// TODO(krzysied): Change it to legit update call
+func UpdateObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind, obj *unstructured.Unstructured) error {
+	if err := DeleteObject(dynamicClient, namespace, name, gvk); err != nil {
```

Comment:
I don't understand this. You should use regular Update call.

Reply:
At this moment, I have no idea how to do this. I mean I know that update method exists. It requires unstructured.Unstructured object which is mostly that same as the one from get function, however it allows only some fields to be overwritten and I haven't figure out yet how to do this.

I would like to proceed  with this solution, cause in my opinion it is not crucial to have a legit update call at this moment. I will fix it later (there is TODO).